### ISSUE TITLE
Time in the panel widget never updates

### ISSUE THREAD

**Issue by mariospr** [projecthamster/hamster-shell-extension#189]
I just installed this extension in GNOME 3.24 (after merging locally PR #174, which is needed too) and found that the time in the panel and the dropdown panel was never being updated.

After some investigation, I found that the GSourceFunc callback installed to be called every 60 secs from panelWidget.js was not explicitly returning any value, meaning that implicitly was returning GLib.SOURCE_REMOVE (false), causing the callback to be executed only once.

I believe that fixing that should fix this problem

**Comment by mariospr** [comment]
This should fix it: https://github.com/projecthamster/hamster-shell-extension/pull/190

**Comment by elbenfreund** [comment]
Would you be willing to try the current develop branch to determine if the issue is solved?

**Comment by mariospr** [comment]
I would, will try to do it as soon as I can possibly do it (not later than Monday)

**Comment by mariospr** [comment]
I updated to the develop branch and I can confirm this particular issue is fixed, since the timer both in the panel and the popup menu now updates every minute as expected, when a task is in progress.

However, and it's surprising I haven't noticed this before, I realized that when I enter a new task, or stop one in progress, from the popup menu I don't get "immediate feedback": I would expect to see the new/stopped task reflected both in the text showing up in the top panel and in the menu item in the popup menu, but instead I see no change until a minute has passed (at which time the callback kicks-in and updates the text).

Is that a known issue already?

**Comment by elbenfreund** [comment]
Hello Mario, thanks for getting back to me. I do not think this is documented so far. It is quite likely a new issue resulting from the restructuring of the codebase compared to ``master``. I do not think many peoply have tested the extension yet, so I would not be surprised if you are the first to notice.

Please feel free to open an issue with the details.

**Comment by mariospr** [comment]
Sure, here you are: https://github.com/projecthamster/hamster-shell-extension/issues/205

**Comment by elbenfreund** [comment]
As this seems sorted I will close this one...
